edx/frontend-app-gradebook
# Add Show Course Staff option and exclude all course roles by default

**TL;DR -** By default, don't show users that have a course role, and add a filter to show users with a course role

JIRA: [EDUCATOR-5571](https://openedx.atlassian.net/browse/EDUCATOR-5571)

**What changed?**

- By default, don't show users that have a course role, and add a filter to show users with a course role
![Screen Shot 2021-04-08 at 3 11 43 PM](https://user-images.githubusercontent.com/1639231/114083281-c5648d00-987c-11eb-86bf-cb2bc389226f.png)


**Developer Checklist**

**Testing Instructions**

This requires https://github.com/edx/edx-platform/pull/26994 to be running locally.
Follow the same setup steps in that PR:

```
-Log in to LMS
-Navigate to a course with users enrolled (the managements command create_test_users may be useful here)
-Under the Instructor Dashboard -> Membership -> Course Team Management, add enrolled learners to the following roles(see below*):
+ Staff
+ Admin
+ Beta Tester
+ Course Data Researcher
```

-Go to the gradebook for that course.
-You should not see any users with course roles.
-Check the "Show Course Staff" box.
-You should now see enrolled users with roles.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:



FYI: @edx/masters-devs-gta
